package/Editor/Isometric Workflow/IsometricWorkflow.cs

```diff
@@ -0,0 +1,300 @@
+                    break;
+                case Base.Custom:
+                    baseTexture = isometricWorkflowUI.customTexture;
+                    break;
+                default:
+                    break;
+            }
+
+            inferenceIdByAssetList.Clear();
+
+            int totalRequests = assetList.Count;
+            int completedRequests = 0;
+
+            foreach (string assetName in assetList)
+            {
+                string tempName = assetName;
+                bool useBaseTexture = baseTexture != null;
+                string modelName = settings.isometricModels.Find(x => x.style == selectedModel).modelData.name;
+                PromptWindow.GenerateImage(modelName, useBaseTexture, useBaseTexture, useBaseTexture ? baseTexture : null, 4, $"isometric, solo, centered, solid color background, {selectedTheme}, {assetName}", 30, 1024, 1024, 6, "-1", useBaseTexture, 0.8f,
+                (inferenceId) =>
+                {
+                    if (inferenceIdByAssetList.ContainsKey(tempName))
+                    {
+                        inferenceIdByAssetList[tempName] = inferenceId;
+                    }
+                    else
+                    { 
+                        inferenceIdByAssetList.Add(tempName, inferenceId);
+                    }
+
+                    completedRequests++;
+                    if (completedRequests == totalRequests)
+                    {
+                        _onRequestSent?.Invoke();
+                    }
+                });
+            }
+        }
+
+        /// <summary>
+        /// Regenerate only on inference.
+        /// </summary>
+        /// <param name="_assetName"> Based on the previous asset name</param>
+        /// <param name="_onRequestSent"> Callback from generation </param>
+        public void RegenerateImages(string _assetName, Action _onRequestSent)
+        {
+            Texture2D baseTexture = null;
+            switch (selectedBase)
+            {
+                case Base.None:
+                    baseTexture = null;
+                    break;
+                case Base.Square:
+                    baseTexture = settings.squareBaseTexture;
+                    break;
+                case Base.Custom:
+                    baseTexture = isometricWorkflowUI.customTexture;
+                    break;
+                default:
+                    break;
+            }
+
+            int totalRequests = assetList.Count;
+            int completedRequests = 0;
+
+            string tempName = _assetName;
+            bool useBaseTexture = baseTexture != null;
+            string modelName = settings.isometricModels.Find(x => x.style == selectedModel).modelData.name;
+            PromptWindow.GenerateImage(modelName, useBaseTexture, useBaseTexture, useBaseTexture ? baseTexture : null, 4, $"isometric, solo, centered, solid color background, {selectedTheme}, {_assetName}", 30, 1024, 1024, 6, "-1", useBaseTexture, 0.8f,
```

Comment:
⚠ The prompt is repeted here

Reply:
Is inside the method to regenerate an inference 🤔